File: bot.js
const { Telegraf, Markup } = require("telegraf");

const bot = new Telegraf("7763395301:AAF3thVNH883Rzmz0RTpsx3wuiCG_VLpa-g");

// Veriler bellekte tutuluyor, restartta sıfırlanır!
const users = {};
const admins = new Set();

// Başlangıç admini buraya koy:
// Botu kullanan ilk kişi veya kendi ID'niz
admins.add(8121637254); // <-- kendi Telegram ID'nizi buraya yazın

// Yardımcı fonksiyonlar
function ensureUser(id) {
  if (!users[id]) {
    users[id] = {
      balance: 0,
      bank: 0,
      lastBonus: 0,
    };
  }
}

function formatMoney(num) {
  return num.toLocaleString() + "₺";
}

function formatBalance(user) {
  return `💰 Cüzdan: ${formatMoney(user.balance)}\n🏦 Banka: ${formatMoney(user.bank)}`;
}

function isAdmin(id) {
  return admins.has(id);
}

function canTakeBonus(user) {
  return Date.now() - user.lastBonus > 24 * 60 * 60 * 1000;
}

// Ana menü butonları
function mainMenu() {
  return Markup.inlineKeyboard([
    [Markup.button.callback("💰 Bakiye", "bakiye"), Markup.button.callback("🎁 Bonus Al", "bonus")],
    [Markup.button.callback("🎟️ Kazı Kazan", "kazi"), Markup.button.callback("🏦 Faiz Al", "faiz")],
    [Markup.button.callback("🕵️‍♂️ Hırsızlık", "hirsizlik"), Markup.button.callback("🎲 Bahis Oyna", "bahis")],
    [Markup.button.callback("🎰 Slot Oyna", "slot"), Markup.button.callback("💳 Banka İşlemleri", "banka")],
    [Markup.button.callback("🏆 Liderlik", "liderlik"), Markup.button.callback("🆔 ID Göster", "id")],
    [Markup.button.callback("💸 Para Gönder", "paragonder"), Markup.button.callback("🛠️ Admin Paneli", "adminpanel")]
  ]);
}

// Başlat
bot.start((ctx) => {
  const id = ctx.from.id;
  ensureUser(id);
  ctx.reply(
    `👋 Merhaba ${ctx.from.first_name}!\n\n` +
    "🎰 Kumar botuna hoşgeldin.\nAşağıdaki menüden işlemlerini seçebilirsin.",
    mainMenu()
  );
});

// Butonlar
bot.action("bakiye", (ctx) => {
  const id = ctx.from.id;
  ensureUser(id);
  ctx.answerCbQuery();
  ctx.editMessageText(`💰 Bakiyen:\n${formatBalance(users[id])}`, mainMenu());
});

bot.action("bonus", (ctx) => {
  const id = ctx.from.id;
  ensureUser(id);
  ctx.answerCbQuery();
  const user = users[id];
  if (canTakeBonus(user)) {
    user.balance += 50000;
    user.lastBonus = Date.now();
    ctx.editMessageText("🎉 50.000₺ bonus hesabına eklendi!", mainMenu());
  } else {
    const wait = 24 * 60 * 60 * 1000 - (Date.now() - user.lastBonus);
    const saat = Math.floor(wait / 3600000);
    const dakika = Math.floor((wait % 3600000) / 60000);
    ctx.editMessageText(`⏳ Bonus sadece 24 saatte bir alınabilir.\nLütfen ${saat} saat ${dakika} dakika bekle.`, mainMenu());
  }
});

bot.action("kazi", (ctx) => {
  const id = ctx.from.id;
  ensureUser(id);
  ctx.answerCbQuery();
  const user = users[id];

  if (user.balance < 1000) return ctx.answerCbQuery("❌ Kazı kazan için en az 1.000₺ bakiyen olmalı.", true);

  user.balance -= 1000;

  if (Math.random() < 0.3) {
    const kazanc = 3000 + Math.floor(Math.random() * 3000);
    user.balance += kazanc;
    ctx.editMessageText(`🎉 Kazı Kazan: Kazandın! +${formatMoney(kazanc)}`, mainMenu());
  } else {
    ctx.editMessageText("😞 Kazı Kazan: Kaybettin.", mainMenu());
  }
});

bot.action("faiz", (ctx) => {
  const id = ctx.from.id;
  ensureUser(id);
  ctx.answerCbQuery();
  const user = users[id];
  if (user.bank <= 0) return ctx.answerCbQuery("❌ Bankanda para yok.", true);

  const faizOrani = 0.02;
  const faiz = Math.floor(user.bank * faizOrani);
  user.bank += faiz;

  ctx.editMessageText(`🏦 Banka faizinden %2 kazandın: +${formatMoney(faiz)}`, mainMenu());
});

bot.action("hirsizlik", (ctx) => {
  const id = ctx.from.id;
  ensureUser(id);
  ctx.answerCbQuery();

  ctx.reply("🔍 Para çalmak istediğin kişinin Telegram ID'sini yaz. İptal için /iptal");

  bot.once("text", (ctx2) => {
    const hedefId = Number(ctx2.message.text);
    if (!hedefId || hedefId === id) return ctx2.reply("❌ Geçersiz kullanıcı ID.");
    ensureUser(hedefId);

    const hedef = users[hedefId];
    const user = users[id];

    if (hedef.balance < 1000) return ctx2.reply("❌ Hedef kişinin yeterli parası yok.");

    const cekilen = Math.min(5000, Math.floor(hedef.balance * 0.3));

    if (Math.random() < 0.5) {
      hedef.balance -= cekilen;
      user.balance += cekilen;
      ctx2.reply(`🎉 Başarılı hırsızlık! ${formatMoney(cekilen)} çaldın.`);
    } else {
      ctx2.reply("❌ Hırsızlık başarısız oldu, dikkatli ol!");
    }
  });
});

bot.action("bahis", (ctx) => {
  const id = ctx.from.id;
  ensureUser(id);
  ctx.answerCbQuery();

  ctx.reply("🎲 Bahis için miktar yaz:");

  bot.once("text", (ctx2) => {
    const miktar = Number(ctx2.message.text);
    if (!miktar || miktar <= 0) return ctx2.reply("❌ Geçersiz miktar.");
    const user = users[id];
    if (user.balance < miktar) return ctx2.reply("❌ Yeterli bakiyen yok.");

    user.balance -= miktar;

    if (Math.random() < 0.5) {
      const kazanclik = miktar * 2;
      user.balance += kazanclik;
      ctx2.reply(`🎉 Bahisi kazandın! +${formatMoney(kazanclik)}`);
    } else {
      ctx2.reply("😞 Bahisi kaybettin.");
    }
  });
});

bot.action("slot", (ctx) => {
  const id = ctx.from.id;
  ensureUser(id);
  ctx.answerCbQuery();

  const user = users[id];
  if (user.balance < 1000) return ctx.answerCbQuery("❌ Slot oynamak için en az 1.000₺ olmalı.", true);

  user.balance -= 1000;

  const semboller = ["🍒", "🍋", "🍉", "🔔", "⭐"];
  const secilen = [];
  for (let i = 0; i < 3; i++) {
    secilen.push(semboller[Math.floor(Math.random() * semboller.length)]);
  }

  let mesaj = `🎰 Slot sonuçları:\n${secilen.join(" | ")}\n`;

  if (secilen[0] === secilen[1] && secilen[1] === secilen[2]) {
    const kazanc = 10000;
    user.balance += kazanc;
    mesaj += `🎉 Tebrikler! Jackpot! +${formatMoney(kazanc)}`;
  } else if (secilen[0] === secilen[1] || secilen[1] === secilen[2] ||

File: bot.py
import json
import random
from datetime import datetime, timedelta
from telegram import Update, ReplyKeyboardMarkup
from telegram.ext import ApplicationBuilder, CommandHandler, ContextTypes

TOKEN = "7763395301:AAF3thVNH883Rzmz0RTpsx3wuiCG_VLpa-g"
DATA_FILE = "data.json"
SABIT_ADMINLER = [8121637254, 987654321]
BAŞLANGIÇ_TL = 10000
BONUS_TL = 50000
BONUS_SÜRE = 86400  # 24 saat

# Ana menü
main_menu = ReplyKeyboardMarkup([
    ["/start", "/bonus"],
    ["/bakiye", "/kazikazan 100"],
    ["/slot 100", "/risk 100"]
], resize_keyboard=True)

# Veri yönetimi
def veri_yükle():
    try:
        with open(DATA_FILE, "r") as f:
            return json.load(f)
    except FileNotFoundError:
        return {}

def veri_kaydet(data):
    with open(DATA_FILE, "w") as f:
        json.dump(data, f, indent=4)

def kullanıcı_kontrol(user_id):
    data = veri_yükle()
    if str(user_id) not in data:
        data[str(user_id)] = {
            "tl": BAŞLANGIÇ_TL,
            "last_bonus": "1970-01-01 00:00:00",
            "admin": False
        }
        veri_kaydet(data)

def admin_mi(user_id):
    data = veri_yükle()
    return user_id in SABIT_ADMINLER or data.get(str(user_id), {}).get("admin", False)

def tl_güncelle(user_id, miktar):
    data = veri_yükle()
    data[str(user_id)]["tl"] += miktar
    veri_kaydet(data)

# Komutlar
async def start(update: Update, context: ContextTypes.DEFAULT_TYPE):
    user_id = update.effective_user.id
    kullanıcı_kontrol(user_id)
    await update.message.reply_text(
        f"Hoş geldin {update.effective_user.first_name}!\nHesabına {BAŞLANGIÇ_TL} TL yüklendi!",
        reply_markup=main_menu
    )

async def bakiye(update: Update, context: ContextTypes.DEFAULT_TYPE):
    user_id = update.effective_user.id
    kullanıcı_kontrol(user_id)
    data = veri_yükle()
    await update.message.reply_text(f"Bakiyen: {data[str(user_id)]['tl']:,} TL")

async def bonus(update: Update, context: ContextTypes.DEFAULT_TYPE):
    user_id = update.effective_user.id
    kullanıcı_kontrol(user_id)
    data = veri_yükle()
    user = data[str(user_id)]
    son = datetime.strptime(user["last_bonus"], "%Y-%m-%d %H:%M:%S")
    now = datetime.now()

    if now - son >= timedelta(seconds=BONUS_SÜRE):
        user["last_bonus"] = now.strftime("%Y-%m-%d %H:%M:%S")
        user["tl"] += BONUS_TL
        msg = f"Günlük bonus alındı! +{BONUS_TL:,} TL"
    else:
        kalan = timedelta(seconds=BONUS_SÜRE) - (now - son)
        msg = f"Bonus zaten alındı!\nYeniden almak için bekle: {str(kalan).split('.')[0]}"

    veri_kaydet(data)
    await update.message.reply_text(msg)

async def kazikazan(update: Update, context: ContextTypes.DEFAULT_TYPE):
    user_id = update.effective_user.id
    kullanıcı_kontrol(user_id)

    try:
        miktar = int(context.args[0])
    except:
        return await update.message.reply_text("Kullanım: /kazikazan [miktar]")

    data = veri_yükle()
    user = data[str(user_id)]

    if user["tl"] < miktar:
        return await update.message.reply_text("Yetersiz bakiye!")

    user["tl"] -= miktar
    if random.randint(1, 100) <= 30:
        kazanç = miktar * 3
        user["tl"] += kazanç
        msg = f"Tebrikler! Kazı Kazan'dan {kazanç:,} TL kazandın!"
    else:
        msg = "Üzgünüm, bu sefer olmadı..."
    
    veri_kaydet(data)
    await update.message.reply_text(msg)

async def risk(update: Update, context: ContextTypes.DEFAULT_TYPE):
    user_id = update.effective_user.id
    kullanıcı_kontrol(user_id)

    try:
        miktar = int(context.args[0])
    except:
        return await update.message.reply_text("Kullanım: /risk [miktar]")

    data = veri_yükle()
    user = data[str(user_id)]

    if user["tl"] < miktar:
        return await update.message.reply_text("Yetersiz bakiye!")

    user["tl"] -= miktar
    if random.randint(1, 100) <= 40:
        kazanç = miktar * 2
        user["tl"] += kazanç
        msg = f"Şanslısın! {kazanç:,} TL kazandın!"
    else:
        msg = "Kaybettin..."

    veri_kaydet(data)
    await update.message.reply_text(msg)

async def slot(update: Update, context: ContextTypes.DEFAULT_TYPE):
    user_id = update.effective_user.id
    kullanıcı_kontrol(user_id)

    try:
        miktar = int(context.args[0])
    except:
        return await update.message.reply_text("Kullanım: /slot [miktar]")

    data = veri_yükle()
    user = data[str(user_id)]

    if user["tl"] < miktar:
        return await update.message.reply_text("Yetersiz bakiye!")

    user["tl"] -= miktar
    if random.randint(1, 100) <= 30:
        kazanç = miktar * 4
        user["tl"] += kazanç
        msg = f"JACKPOT! {kazanç:,} TL kazandın!"
    else:
        msg = "Slot kaybettin..."

    veri_kaydet(data)
    await update.message.reply_text(msg)

async def admin(update: Update, context: ContextTypes.DEFAULT_TYPE):
    user_id = update.effective_user.id
    if not admin_mi(user_id):
        return await update.message.reply_text("Yetkisiz erişim.")

    try:
        hedef_id = int(context.args[0])
    except:
        return await update.message.reply_text("Kullanım: /admin [id]")

    data = veri_yükle()
    kullanıcı_kontrol(hedef_id)
    data[str(hedef_id)]["admin"] = True
    veri_kaydet(data)
    await update.message.reply_text(f"{hedef_id} artık admin!")

async def parabasma(update: Update, context: ContextTypes.DEFAULT_TYPE):
    user_id = update.effective_user.id
    if not admin_mi(user_id):
        return await update.message.reply_text("Sadece adminler para basabilir!")

    try:
        hedef_id = int(context.args[0])
        miktar = int(context.args[1])
    except:
        return await update.message.reply_text("Kullanım: /parabasma [id] [miktar]")

    kullanıcı_kontrol(hedef_id)
    tl_güncelle(hedef_id, miktar)
    await update.message.reply_text(f"{hedef_id} kişisine {miktar:,} TL basıldı!")

async def paragönder(update: Update, context: ContextTypes.DEFAULT_TYPE):
    user_id = update.effective_user.id

    try:
        hedef_id = int(context.args[0])
        miktar = int(context.args[1])
    except:
        return await update.message.reply_text("Kullanım: /paragönder [id] [miktar]")

    data = veri_yükle()
    kullanıcı_kontrol(hedef_id)

    if data[str(user_id)]["tl"] < miktar:
        return await update.message.reply_text("Yetersiz bakiye!")

    data[str(user_id)]["tl"] -= miktar
    data[str(hedef_id)]["tl"] += miktar
    veri_kaydet(data)
    await update.message.reply_text(f"{miktar:,} TL gönderildi!")

async def id(update: Update, context: ContextTypes.DEFAULT_TYPE):
    if not update.message.reply_to_message:
        return await update.message.reply_text("Lütfen bir mesaja yanıt ver.")
    hedef = update.message.reply_to_message.from_user
    kullanıcı_kontrol(hedef.id)
    data = veri_yükle()
    await update.message.reply_text(f"{hedef.first_name} bakiyesi: {data[str(hedef.id)]['tl']:,} TL")

async def top(update: Update, context: ContextTypes.DEFAULT_TYPE):
    data = veri_yükle()
    sıralama = sorted(data.items(), key=lambda x: x[1]["tl"], reverse=True)[:10]
    metin = "\n".join([f"{i+1}. {uid} - {veri['tl']:,} TL" for i, (uid, veri) in enumerate(sıralama)])
    await update.message.reply_text(f"🏆 En Zenginler:\n{metin}")

# Bot başlatma
if __name__ == "__main__":
    app = ApplicationBuilder().token(TOKEN).build()
    app.add_handler(CommandHandler("start", start))
    app.add_handler(CommandHandler("bakiye", bakiye))
    app.add_handler(CommandHandler("bonus", bonus))
    app.add_handler(CommandHandler("kazikazan", kazikazan))
    app.add_handler(CommandHandler("risk", risk))
    app.add_handler(CommandHandler("slot", slot))
    app.add_handler(CommandHandler("admin", admin))
    app.add_handler(CommandHandler("parabasma", parabasma))
    app.add_handler(CommandHandler("paragonder", paragonder))
    app.add_handler(CommandHandler("id", id))
    app.add_handler(CommandHandler("top", top))

    print("Bot çalışıyor...")
    app.run_polling()
